sqlx-core/src/postgres/connection/executor.rs

```diff
@@ -118,23 +142,29 @@ impl PgConnection {
         Ok(())
     }
 
-    async fn prepare(&mut self, query: &str, arguments: &PgArguments) -> Result<u32, Error> {
-        if let Some(statement) = self.cache_statement.get_mut(query) {
-            return Ok(*statement);
+    async fn prepare(
+        &mut self,
+        query: &str,
+        arguments: &PgArguments,
+    ) -> Result<&mut PgStatement, Error> {
+        let contains = self.cache_statement.contains_key(query);
+
+        if contains {
+            return Ok(self.cache_statement.get_mut(query).unwrap());
```

Comment:
Same double-lookup here.

Reply:
This we might be able to optimize. The `get_mut` takes a mutable reference to the cache and it's a bit tricky to call it again if you have a mutable reference to it in scope.